### ISSUE TITLE
Model causing bad_access

### ISSUE THREAD

**Issue by dannyvra** [realm/realm-cocoa#2528]
I get a EXC_BAD_ACCESS in Object.swift on line 305 (let mirror = reflection[i].1)

It seems the problem comes from this line in my code     let customers = List<Customer>()

Any ideas why this happens?

Here is part of my model:

```
class Booking: Object {
    
    enum BookingState: Int {
        case Active = 0
    }
    
    dynamic var attendeesLimit: Int = 0
    dynamic var date: NSDate = NSDate()
    dynamic var duration: Int = 0
    dynamic var bookingId: String = ""
    dynamic var calendarId: String = ""

    let customers = List<Customer>()
    dynamic var profileId: String = ""
    dynamic var serviceId: String = ""
    dynamic var notes: String = ""
    dynamic var notifiedCustomer: Bool = false
    dynamic var notifiedEmployee: Bool = false
    var state: BookingState = .Active
    dynamic var title: String = ""
    dynamic var category: BookingCategory?
    
    override static func primaryKey() -> String? {
        return "bookingId"
    }
}

// Customers
class Customer: Object {
    dynamic var customerId: String = ""
    dynamic var birthday: String = ""
    dynamic var city: String = ""
    dynamic var email: String = ""
    dynamic var firstName: String = ""
    dynamic var gender: String = ""
    dynamic var lastName: String = ""
    dynamic var notes: String = ""
    dynamic var phoneCountry: String = ""
    dynamic var phoneNumber: String = ""
    dynamic var street: String = ""
    dynamic var streetNumber: String = ""
    dynamic var type: String = ""
    dynamic var zipCode: String = ""
    
    override static func primaryKey() -> String? {
        return "customerId"
    }
}
```
Thank you!

**Comment by tgoyne** [comment]
The problem appears to be `dynamic var category: BookingCategory?`, as it crashes for me unless I comment that out (I added a minimal `Object` subclass for that, as it's not in the code snippet). Making it explicitly default to `nil` doesn't help.

**Comment by tgoyne** [comment]
This seems to be yet another Swift reflection bug since `Array(Mirror(reflecting: object).children)` also hits the same crash.

**Comment by jpsim** [comment]
I filed a radar about this: http://www.openradar.me/22866270

**Comment by jpsim** [comment]
Closing this issue as it's entirely out of our hands. Feel free to dupe the radar to encourage Apple to address this issue.

**Comment by jpsim** [comment]
FYI this was fixed as of Xcode 7.3 beta (7D111g).
